storage/src/main/java/tech/pegasys/teku/storage/protoarray/ProtoArray.java

```diff
@@ -241,7 +241,7 @@ private Optional<ProtoNode> findHead(
 
     // Perform a sanity check that the node is indeed valid to be the head.
     if (!nodeIsViableForHead(bestNode) && !bestNode.equals(justifiedNode)) {
-      throw new RuntimeException("ProtoArray: Best node is not viable for head");
+      throw new IllegalStateException("ProtoArray: Best node is not viable for head");
```

Comment:
is it worth adding block root and slot here?

Reply:
Seems like a good idea yeah.